install

```diff
@@ -198,13 +207,26 @@ sudo "/bin/chmod", "g+rwx", HOMEBREW_CACHE if chmod? HOMEBREW_CACHE
 sudo "/usr/sbin/chown", ENV['USER'], HOMEBREW_CACHE if chown? HOMEBREW_CACHE
 sudo "/usr/bin/chgrp", "admin", HOMEBREW_CACHE if chgrp? HOMEBREW_CACHE
 
-if macos_version >= "10.9"
-  developer_dir = `/usr/bin/xcode-select -print-path 2>/dev/null`.chomp
-  if developer_dir.empty? || !File.exist?("#{developer_dir}/usr/bin/git")
+if should_install_command_line_tools
+  ohai "Searching online for the Command Line Tools"
+  # This temporary file prompts the 'softwareupdate' utility to list the Command Line Tools
+  clt_placeholder = "/tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress"
+  sudo "/usr/bin/touch", clt_placeholder
+  clt_label = `softwareupdate -l | grep -B 1 -E "Command Line (Developer|Tools)" | awk -F"*" '/^ +\\*/ {print $2}' | sed 's/^ *//' | head -n1`.chomp
+  ohai "Installing #{clt_label}"
+  sudo "/usr/sbin/softwareupdate", "-i", clt_label
+  sudo "/bin/rm", "-f", clt_placeholder
+end
+
+# Headless install may have failed, so fallback to original 'xcode-select' method
+if should_install_command_line_tools
+  if STDIN.tty?
     ohai "Installing the Command Line Tools (expect a GUI popup):"
     sudo "/usr/bin/xcode-select", "--install"
     puts "Press any key when the installation has completed."
     getc
+  else
+    abort "Unable to proceed with manual Command Line Tools install without user input."
```

Comment:
In my perfect world none of them would exceed that but the important thing is that the string `Error: unable to proceed with manual Command Line Tools install without user input!` is less than 80 characters as Bash's prose wrapping leaves a bit to be desired.